### PR TITLE
docs: release notes for 1.0 revisions

### PR DESCRIPTION
**Pull request by majorgreys**
Aside from some minor edits, the preface has been redone to separate out different types of changes and actions that a library user needs to take when migrating.

📝 [Preview](https://output.circle-artifacts.com/output/job/dc29a8ee-5d02-41e0-b9a4-44dc6d00744d/artifacts/0/tmp/docs/_release_notes_all.html)

### CODE REVIEW

File: releasenotes/notes/release-1.0-758fc917e40d7eb3.yaml [hunk 1]
```diff
@@ -15,69 +15,207 @@ prelude: >
...
+    :header-rows: 1
+    :widths: 45, 45, 10
+
+    * - Variable
+      - Replacement
+      - Note
+    * - ``DATADOG_`` prefix
+      - ``DD_`` prefix
+      - :ref:`📝<remove-datadog-envs>`
+    * - ``DATADOG_SERVICE_NAME`` prefix
```
Comment (brettlangdon) [inline releasenotes/notes/release-1.0-758fc917e40d7eb3.yaml:51]:
```suggestion
    * - ``DATADOG_SERVICE_NAME``
```

File: releasenotes/notes/release-1.0-758fc917e40d7eb3.yaml [hunk 2]
```diff
@@ -15,69 +15,207 @@ prelude: >
...
+      - ``DD_SERVICE`` prefix
```
Comment (brettlangdon) [inline releasenotes/notes/release-1.0-758fc917e40d7eb3.yaml:52]:
```suggestion
      - ``DD_SERVICE``
```

File: releasenotes/notes/release-1.0-758fc917e40d7eb3.yaml [hunk 3]
```diff
@@ -15,69 +15,227 @@ prelude: >
...
+  .. list-table::
+    :header-rows: 1
+    :widths: 45, 45, 10
+
+    * - Variable
+      - Replacement
+      - Note
+    * - ``DATADOG_`` prefix
+      - ``DD_`` prefix
+      - :ref:`📝<remove-datadog-envs>`
```
Comment (majorgreys) [inline releasenotes/notes/release-1.0-758fc917e40d7eb3.yaml:54]: Preferably we should use `|:memo:|` provided by `sphinxemoji` but the replacement does not seem to work within a `:ref:` context.

By the looks of it, the emoji short code should be replaced before the cross-reference link is transformed:

https://docutils.sourceforge.io/docs/ref/transforms.html#transform-priority-range-categories

https://github.com/sphinx-contrib/emojicodes/blob/c536aee39972a4dfa6d2111120f50411b8fd3bee/sphinxemoji/sphinxemoji.py#L49

https://github.com/sphinx-doc/sphinx/blob/4221d1a5163d023d80a34bd4a12e9c2a2c648ae3/sphinx/transforms/references.py#L30

File: releasenotes/notes/release-1.0-758fc917e40d7eb3.yaml [hunk 4]
```diff
@@ -3,7 +3,7 @@ prelude: >
   The Datadog APM Python team is happy to announce the release of v1.0.0 of
   ddtrace. This release simplifies the public
   :ref:`interface<versioning_interfaces>` and introduces a formal
-  :ref:`versioning policy<versioning>`. These changes unlock performance
+  :ref:`versioning policy<versioning>`. These changes future unlock performance
```
Comment (majorgreys) [inline releasenotes/notes/release-1.0-758fc917e40d7eb3.yaml:6]:
```suggestion
  :ref:`versioning policy<versioning>`. These changes unlock future performance
```

File: releasenotes/notes/release-1.0-758fc917e40d7eb3.yaml [hunk 5]
```diff
@@ -15,69 +15,227 @@ prelude: >
...
+      - :ref:`📝<remove-span-types-enum>`
+    * - ``ddtrace.helpers``
+      - ``get_correlation_ids``
+      - :ref:`📝<remove-helpers>`
+    * - ``ddtrace.settings``
+      - ``Config.HTTPServerConfig``
+      - :ref:`📝<remove-config-httpserver>`
+
+
+  Removed legacy integration tracing
```
Comment (brettlangdon) [inline releasenotes/notes/release-1.0-758fc917e40d7eb3.yaml:161]: I think these are missing: `releasenotes/notes/update-deprecated-contrib-removal-version-ca920d9e1d91b6e0.yaml`

**Review by brettlangdon — COMMENTED** [review]
one note, and I think we can omit the span.get_tags/metrics feature note.

**Comment by brettlangdon** [comment]
I think we still want to add a section either to the prelude or maybe the upgrading doc about the change to logging, that we don't call basic config by default anymore, what to do to get the old behavior back.

File: releasenotes/notes/release-1.0-758fc917e40d7eb3.yaml [hunk 6]
```diff
@@ -1,83 +1,255 @@
...
+  and incompatible changes to the public interface.
 
 
   The v1.0.0 release is an important milestone for the library as it has grown
   substantially in scope. The first commit to the library was made on June 20,
-  2016. Nearly sixty minor releases later, the library now includes over 60
-  integrations for libraries and has expanded from tracing to
-  support the Continuous Profiler and CI Visibility.
+  2016. Nearly sixty minor releases later, the library now includes over sixty
+  integrations for libraries. And has expanded from Tracing to support the
```
Comment (majorgreys) [inline releasenotes/notes/release-1.0-758fc917e40d7eb3.yaml:13]:
```suggestion
  integrations for libraries. And the library has expanded from Tracing to support the
```

**Review by brettlangdon — DISMISSED** [review]

**Review by mabdinur — DISMISSED** [review]

**Review by brettlangdon — DISMISSED** [review]

File: docs/versioning.rst [hunk 7]
```diff
@@ -15,16 +15,41 @@ Release support
...
+
+.. list-table::
+   :header-rows: 1
+
+   * - Level
+     - Policy
+
+       .. _versioning_support_ga:
+   * - General Availability (GA)
+     - Full implementation of all features. Full support for new features, bug and security fixes.
```
Comment (Kyle-Verhoog) [inline docs/versioning.rst:40]: I'm not 100% sure if "full" is well defined here or what "Full implementation of all features" means

**Review by Kyle-Verhoog — COMMENTED** [review]
Just one question/concern but otherwise lgtm!

**Review by Kyle-Verhoog — DISMISSED** [review]

**Review by brettlangdon — APPROVED** [review]

**Pull request merged**